src/message/fetchActions.js

```diff
@@ -82,8 +82,14 @@ export const fetchMessages = (
   dispatch(backgroundFetchMessages(narrow, anchor, numBefore, numAfter, useFirstUnread));
 };
 
-export const fetchMessagesAtFirstUnread = (narrow: Narrow): Action =>
-  fetchMessages(narrow, 0, config.messagesPerRequest / 2, config.messagesPerRequest / 2, true);
+export const fetchMessagesAtFirstUnread = (narrow: Narrow, anchor: number): Action =>
```

Comment:
You are changing the meaning of the function!
When you provide an anchor, it is not `at first unread`.
The `fetchMessages` already does what you want from this function.